frontend/src/app/word-models/word-models.component.ts

```diff
@@ -60,31 +62,48 @@ export class WordModelsComponent implements DoCheck, OnInit {
         }
     }
 
-    async ngOnInit(): Promise<void> {
+    async initialize(): Promise<void> {
         this.user = await this.authService.getCurrentUserPromise();
         this.corpusService.currentCorpus.subscribe(this.setCorpus.bind(this));
     }
 
+    teardown(): void {
+        this.setParams({
+            query: null,
+            show: null
+        });
+    }
+
+    setStateFromParams(params: Params) {
+        if (params.has('query')) {
+            this.queryText = params.get('query');
+            this.activeQuery = this.queryText;
+            this.validateQuery();
+            if (this.queryFeedback === undefined) {
+                this.wordModelsService
+                    .wordInModel(this.queryText, this.corpus.name)
+                    .then(this.handleWordInModel.bind(this))
+                    .catch(() => (this.queryFeedback = { status: 'error' }));
+            }
+            this.activeQuery = this.queryText;
+            this.queryFeedback = { status: 'success' };
+        }
```

Comment:
Similar to above: if the parameters do not have a `query` key, you should not do nothing, but set the state to "no query".